wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/WorkspaceDao.java

```diff
@@ -133,13 +133,13 @@
     List<WorkspaceImpl> getByNamespace(String namespace) throws ServerException;
 
     /**
-     * Gets list of workspaces where user is worker
+     * Gets list of workspaces where given user is worker and has read permission
```

Comment:
It should get all workspaces by relation between user and workspace. Take a look implementation in codenvy project https://github.com/codenvy/codenvy/blob/workspacePermissions/core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/WorkspaceDaoImpl.java#L264-264